### PR TITLE
refactor: use readcontracts for xsushi bar data

### PR DESCRIPTION
**Pull request by danielr18**
**Notes:**

- The `staleTime` of the graphql query is 30 seconds. I used the same time for the contract data, and refresh it every 30 seconds.

### CODE REVIEW

**Comment by danielr18** [comment]
The RPC calls are failing with `Method not allowed for this key`.

**Comment by LufyCZ** [comment]
> The RPC calls are failing with `Method not allowed for this key`.

Our public key has the `eth_getBalance` and `eth_getLogs` method disabled due to abuse. Don't think either of those are being used here though.

Actually, `useBalancesWeb3` might be trying to fetch the native balance with `eth_getBalance`, if you get that disable, the error should clear up.

We use our api for balance fetching now though, you can have a look at how it's done in the swap widget.

**Comment by danielr18** [comment]
Switched the implementation to just use `useReadContracts` instead of custom hooks. I think they did more than what was needed for this case.

I looked at the `BalanceProvider` but that keeps the balances for the connected account.

**Comment by LufyCZ** [comment]
> Switched the implementation to just use `useReadContracts` instead of custom hooks. I think they did more than what was needed for this case.
> 
> I looked at the `BalanceProvider` but that keeps the balances for the connected account.

Yeah, you're right, I didn't think about fetching balances for non-connected accounts when writing that thing. If `useReadContracts` works, then I don't think there's a better solution 👍

File: apps/web/src/ui/stake/SushiBarProvider.tsx [hunk 1]
```diff
@@ -19,34 +29,68 @@ export const SushiBarProvider: FC<{
...
+        return [
+          Amount.fromRawAmount(SUSHI[ChainId.ETHEREUM], data[0]),
+          Amount.fromRawAmount(XSUSHI[ChainId.ETHEREUM], data[1]),
+        ] as const
+      },
+      staleTime: 30000,
+    },
+  })
+
+  useWatchByInterval({ key: balanceAndSupplyQueryKey, interval: 30000 })
```
Comment (0xMasayoshi) [inline apps/web/src/ui/stake/SushiBarProvider.tsx:71]: any reason to use useWatchByInterval instead of setting `refetchInterval` on useReadContracts?

Comment (danielr18) [reply apps/web/src/ui/stake/SushiBarProvider.tsx]: No. I've changed it to use `refetchInterval` now.

**Pull request merged**